include/cmm/Node.h

```diff
@@ -12,6 +12,7 @@
 
 // Our includes
 #include <cmm/Types.h>
+#include <cmm/Visitor.h>
```

Comment:
Can we just use a forward declaration? Do we actually need to include this?

Reply:
This is actually needed because of the use of 'VisitorResult'.  No further changes needed.